include/seqan3/search/configuration/mode.hpp

```diff
@@ -6,97 +6,109 @@
 // -----------------------------------------------------------------------------------------------------
 
 /*!\file
- * \brief Provides the mode configuration to define the search modes "all", "all_best", "best" and "strata".
+ * \brief Provides the mode configuration to define the search modes "hit_strata", "hit_all",
+ *        "hit_all_best", "hit_single_best".
  * \author Christopher Pockrandt <christopher.pockrandt AT fu-berlin.de>
  * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
  */
 
 #pragma once
 
-#include <seqan3/core/algorithm/configuration.hpp>
+#include <variant>
+
 #include <seqan3/core/algorithm/pipeable_config_element.hpp>
-#include <seqan3/core/detail/strong_type.hpp>
-#include <seqan3/core/type_traits/basic.hpp>
+#include <seqan3/core/detail/empty_type.hpp>
+#include <seqan3/core/detail/test_accessor.hpp>
+#include <seqan3/core/type_list/traits.hpp>
+#include <seqan3/core/type_traits/template_inspection.hpp>
 #include <seqan3/search/configuration/detail.hpp>
 
-
 namespace seqan3::detail
 {
 
-//!\brief Type for the "all" value for the configuration element "mode".
-//!\ingroup search_configuration
-struct search_mode_all {};
-//!\brief Type for the "all_best" value for the configuration element "mode".
-//!\ingroup search_configuration
-struct search_mode_all_best {};
-//!\brief Type for the "best" value for the configuration element "mode".
-//!\ingroup search_configuration
-struct search_mode_best {};
-
-} // namespace seqan3::detail
-
-namespace seqan3::search_cfg
+/*!\brief Configuration element to receive all hits within the error bounds.
+ * \ingroup search_configuration
+ */
+struct hit_all_tag : public pipeable_config_element<hit_all_tag, empty_type>
 {
+    //!\privatesection
+    //!\brief Internal id to check for consistent configuration settings.
+    static constexpr detail::search_config_id id{detail::search_config_id::mode};
```

Comment:
```suggestion
    static constexpr detail::search_config_id id{detail::search_config_id::hit_strategy};
```
I am wondering if we should not rename this, since mode can be anything. Or just `hit`.

Reply:
I guess it won't clash with the dynamic selector `hit` as it is a scoped enum. I guess then it is consistent to name the id after the common prefix of the static elements.